Revert "feat(CLOUDDST-25034): run rh-sign-image-cosign signing in par…

mmalina opened the pull request:
…allel (#701)"

This reverts commit b820c4637ffb5c7af88cdd24f5cad311e79d8767.

It seems the new version doesn't work as expected. A user reported issues.

## Describe your changes

This was found in the taskrun log:

```
+++ cosign verify --rekor-url=https://rekor-server-trusted-artifact-signer.apps.rosa.rekor-stage.ic5w.p3.openshiftapps.com quay.io/redhat-pending/rhtpa----rhtpa-guac-rhel9:1.2.2-1736761596
Error: --certificate-identity or --certificate-identity-regexp is required for verification in keyless mode
main.go:74: error during command execution: --certificate-identity or --certificate-identity-regexp is required for verification in keyless mode
```

And also:
```
+ cosign -t 3m0s sign '-y --rekor-url=https://rekor-server-trusted-artifact-signer.apps.rosa.rekor-stage.ic5w.p3.openshiftapps.com' --key awskms:///arn:aws:kms:us-east-1:095921525512:key/b2c7a225-fe17-464c-b817-56f2c4c97c69 --sign-container-identity registry.stage.redhat.io/rhtpa/rhtpa-guac-rhel9:1.2.2 quay.io/redhat-pending/rhtpa----rhtpa-guac-rhel9@sha256:3f92720dc115a9ecb5ec75188dd776667d44aad98fab7b40ecdc33131f2a4356
WARNING: the -y --rekor-url=https://rekor-server-trusted-artifact-signer.apps.rosa.rekor-stage.ic5w.p3.openshiftapps.com flag is deprecated and will be removed in a future release. Please use the --y --rekor-url=https://rekor-server-trusted-artifact-signer.apps.rosa.rekor-stage.ic5w.p3.openshiftapps.com flag instead.
Error: unknown flag: --y --rekor-url
main.go:74: error during command execution: unknown flag: --y --rekor-url
```

## Relevant Jira

## Checklist before requesting a review
- [ ] I have marked as draft or added `do not merge` label if there's a dependency PR
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [ ] My commit message includes `Signed-off-by: My name <email>`
- [ ] I have bumped the task/pipeline version string and updated changelog in the relevant README
- [ ] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)

mmalina committed:
Revert "feat(CLOUDDST-25034): run rh-sign-image-cosign signing in parallel (#701)"

This reverts commit b820c4637ffb5c7af88cdd24f5cad311e79d8767.

It seems the new version doesn't work as expected. A user reported
issues.

johnbieren commented:
Did you bring this up with Jindrich? How did the pr pass e2e? I have also had successful e2e runs since this merged

mmalina commented:
> Did you bring this up with Jindrich? 

Yes, I pinged him in the support thread and he acknowledged that it's missing something.

> How did the pr pass e2e? I have also had successful e2e runs since this merged

I think e2e doesn't use cosign signing - @jinqi7 can you confirm?

johnbieren commented:
> > Did you bring this up with Jindrich?
> 
> Yes, I pinged him in the support thread and he acknowledged that it's missing something.
> 
Did he say he would open a new PR adding it back but properly? I was pinged by his manager last week to make sure this code got in, so I don't want us to just revert it and forget about it.

> > How did the pr pass e2e? I have also had successful e2e runs since this merged
> 
> I think e2e doesn't use cosign signing - @jinqi7 can you confirm?

It should... the cosign-secret is defined in e2e https://github.com/konflux-ci/e2e-tests/blob/main/tests/release/pipelines/rh_advisories.go#L254

Also his PR failed e2e until we added the new PUBLIC_KEY value to the secret

mmalina commented:
> > > Did you bring this up with Jindrich?
> > 
> > 
> > Yes, I pinged him in the support thread and he acknowledged that it's missing something.
> 
> Did he say he would open a new PR adding it back but properly? I was pinged by his manager last week to make sure this code got in, so I don't want us to just revert it and forget about it.

The Jira was still in Review. I added a comment about the revert: https://issues.redhat.com/browse/CLOUDDST-25034

So I believe that should make it clear that it cannot be resolved yet until a new PR with improved code is merged.
 
> > > How did the pr pass e2e? I have also had successful e2e runs since this merged
> > 
> > 
> > I think e2e doesn't use cosign signing - @jinqi7 can you confirm?
> 
> It should... the cosign-secret is defined in e2e https://github.com/konflux-ci/e2e-tests/blob/main/tests/release/pipelines/rh_advisories.go#L254
> 
> Also his PR failed e2e until we added the new PUBLIC_KEY value to the secret

Hmm, then I don't really know how it passed. @midnightercz any idea how it would work in e2e but not for the user?
